app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultStateProducer.kt

```diff
@@ -32,9 +33,14 @@ class ShortCodeSearchResultStateProducer(
               }
         }
 
+    val viewPatient = events.ofType<ViewPatient>()
+        .doOnNext { ui.openPatientSummary(it.patientUuid) }
+        .flatMap { Observable.empty<ShortCodeSearchResultState>() }
```

Comment:
Correct, 2 things.

1. All state changes are currently observed on the Ui thread. Since this side-effect is supposed to be executed on the Ui thread, this isn't a problem.

2. Again, once we move to Mobius, these effects will be handled by the effect handler on a thread that is specified explicitly.

Reply:
What I meant was that the `doOnNext` itself won't be executed on the main thread, it will be executed on whatever thread the chain will run in.

Reply:
The chain will run on the main thread because, the incoming event is from the UI.